activation/identity_states.go

```diff
@@ -130,3 +138,44 @@ func (s *IdentityStateStorage) All() map[types.NodeID][]IdentityStateInfo {
 	defer s.mu.RUnlock()
 	return s.identities
 }
+
+func (s *IdentityStateStorage) SetEligibilities(
+	id types.NodeID,
+	eligibilities map[types.LayerID][]types.VotingEligibility,
+) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.eligibilities[id] = eligibilities
+}
+
+func (s *IdentityStateStorage) SetProposals(id types.NodeID, proposal *types.Proposal) {
```

Comment:
There's a bit of inconsistency, but I'm not sure what the requirements are. `SetEligibilities` overwrites proofs for the given ID. Basically, `s.eligibilities` holds proofs for the current epoch. On the other hand, `SetProposals` adds a proposal (maxing at 100 when it starts to drop the oldest) so it can potentially keep many epochs of proposals in it.

Also, the name of `SetProposals` method is a little misleading. It _adds_ or _records_ a proposal. _Set_ suggests that whatever was set previously is overwritten, which is not the case.